src/Endpoints/Indexes.php

```diff
@@ -152,49 +157,26 @@ public function fetchInfo(): self
         return $this->fill($response);
     }
 
-    public function update($body): self
+    public function update($body): array
     {
-        $response = $this->http->put(self::PATH.'/'.$this->uid, $body);
-
-        return $this->fill($response);
+        return $this->http->put(self::PATH.'/'.$this->uid, $body);
     }
 
     public function delete(): array
     {
         return $this->http->delete(self::PATH.'/'.$this->uid) ?? [];
     }
 
-    // Updates
+    // Tasks
 
-    public function getUpdateStatus($updateId): array
+    public function getTask($uid): array
     {
-        return $this->http->get(self::PATH.'/'.$this->uid.'/updates/'.$updateId);
+        return $this->tasks->getIndexTask(self::PATH.'/'.$this->uid, $uid);
     }
 
-    public function getAllUpdateStatus(): array
+    public function getAllTasks(): array
     {
-        return $this->http->get(self::PATH.'/'.$this->uid.'/updates');
-    }
-
-    /**
-     * @param string $updateId
-     * @param int    $timeoutInMs
-     * @param int    $intervalInMs
-     *
-     * @throws TimeOutException
-     */
-    public function waitForPendingUpdate($updateId, $timeoutInMs = 5000, $intervalInMs = 50): array
-    {
-        $timeout_temp = 0;
-        while ($timeoutInMs > $timeout_temp) {
-            $res = $this->getUpdateStatus($updateId);
-            if ('enqueued' != $res['status'] && 'processing' != $res['status']) {
-                return $res;
-            }
-            $timeout_temp += $intervalInMs;
-            usleep(1000 * $intervalInMs);
-        }
-        throw new TimeOutException();
+        return $this->tasks->allIndexTask(self::PATH.'/'.$this->uid);
```

Comment:
to be as consistent as possible with the other routes I would suggest the following naming: 

```
  getClientTask
  getClientTasks
  getIndexTask
  getIndexTasks
```